ddtrace/context.py

```diff
@@ -140,8 +147,15 @@ def get(self):
         """
         with self._lock:
             if self._is_finished():
+                # get the trace
                 trace = self._trace
                 sampled = self._sampled
+                sampling_priority = self._sampling_priority
+                # attach the sampling priority to the spans
+                if sampled and sampling_priority is not None:
+                    for span in trace:
+                        span.set_metric(SAMPLING_PRIORITY_KEY, sampling_priority)
```

Comment:
should we attach this metric to all spans or only to the root? Asking because this operation is executed in the main thread that is serving the application.

Reply:
Ideally, only the root. But I'm not sure we can reliably find it here with the exact same algorithm as the Agent?

Reply:
technically it's a list, so they are ordered with the first item being always the root span (at least of this tracing context)... there are case that you have in mind that this is not true?